ext/session/session.c

```diff
@@ -710,6 +710,23 @@ static PHP_INI_MH(OnUpdateCookieLifetime) /* {{{ */
 }
 /* }}} */
 
+static PHP_INI_MH(OnUpdateCacheExpire)
+{
+	SESSION_CHECK_ACTIVE_STATE;
+	SESSION_CHECK_OUTPUT_STATE;
+
+#ifdef ZEND_ENABLE_ZVAL_LONG64
+	const zend_long maxexpire = ((ZEND_LONG_MAX - INT_MAX) / 60) - 1;
+#else
+	const zend_long maxexpire = ((ZEND_LONG_MAX / 2) / 60) - 1;
+#endif
```

Comment:
It seems to me that this is highly platform dependent, since the C standard makes no claim about `time_t` other than that it is an arithmetic type (could even be float/double). Current POSIX states "time_t shall be an integer type with a width of at least 64 bits". Now, that it not true for our x86 Windows builds (it's 32bit wide there); I don't know about other systems.

To handle this cleanly, we should probably determine something like `TIME_MIN` and `TIME_MAX` during configuration (and deal one way or another with non integral `time_t` implementations).

Reply:
Concretally, even QNX does not implement time_t as float/double. Only one os I know does differently, vxworks. But not exactly the kind of os you deploy php :-) anyhow I doubt very much it s building in it.